src/passes/OptimizeInstructions.cpp

```diff
@@ -1391,6 +1391,19 @@ struct OptimizeInstructions
       right->type = Type::i32;
       return right;
     }
+    {
+      // ~(1 << x) aka (1 << x) ^ -1  ==>  rotl(-2, x)
+      if (matches(curr,
+                  binary(Abstract::Xor,
+                         binary(&curr, Abstract::Shl, ival(1), any()),
+                         ival(-1)))) {
+        // reuse "curr" and "right" expressions
```

Comment:
I don't think this comment is necessary, since most of these transformations are also doing this.